src/hooks/useAllowList.ts

```diff
@@ -1,13 +1,21 @@
 import { useLogin } from './useLogin';
 
 type AllowListType = 'arbitrum' | 'rwa' | 'none';
-export function useAllowList(): {
-    isAllowed: boolean;
-    allowListType: AllowListType;
-} {
+export function useAllowList():
+    | {
+          isAllowed: boolean;
+          allowListType: AllowListType;
+      }
+    | undefined {
     const { user, status } = useLogin();
 
-    const arbitrumAllowListEnvString = import.meta.env.PH_CONNECT_ARBITRUM_ALLOW_LIST;
+    // if the user is not yet loaded then wait
+    if (user === undefined) {
+        return undefined;
```

Comment:
Right at startup, we don't know if the user is logged in or not. If an user is set in local storage, Connects does a request to Renown to check if it's still valid.

I've changed the hooks (useUserPermissions, etc) that rely on the user to return undefined while this initial load is happening.
This way, the consumers of these hooks can select the best default behaviour for their use case.

For example, when loading the default drives on startup, they are loaded as read drives or normal sync drives according to the user's access level.

If, on startup, the user is not loaded yet but the drive is already synced , we don't assume the user doesn't have access and don't replace the sync drive with a read drive. This avoids deleting and re-adding a drive everytime the page is refreshed.
If when the user is loaded and we detect that he no longer has write access to a synced drive, we then remove it.